src/web/core/src/main/java/org/geoserver/web/data/store/panel/FileModel.java

```diff
@@ -50,8 +51,14 @@ private boolean isSubfile(File root, File selection) {
         return isSubfile(root, selection.getParentFile());
     }
 
+    @Override
     public String getObject() {
-        return delegate.getObject();
+        Object obj = delegate.getObject();
```

Comment:
I was looking at how the code could have possibly compiled before, looks like delegate is a IModel<String>, but in some cases it actually contains a URL? Which is however wrong since StoreInfo.getConnectionParameters returns a Map<String, Serializable>...